src/commands/importService.ts

```diff
@@ -0,0 +1,96 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ProgressLocation, window } from "vscode";
+import { IActionContext } from 'vscode-azureextensionui';
+import * as Constants from '../constants';
+import { ServiceTreeItem } from '../explorer/ServiceTreeItem';
+import { ext } from '../extensionVariables';
+import { localize } from '../localize';
+import { azUtils } from '../utils/azUtils';
+import { cpUtils } from '../utils/cpUtils';
+import { FileDownloader} from '../utils/FileDownloader';
+import { askFolder } from '../utils/vscodeUtils';
+
+export async function importService(context: IActionContext, node?: ServiceTreeItem): Promise<void> {
+    if (!node) {
+        node = <ServiceTreeItem>await ext.tree.showTreeItemPicker(ServiceTreeItem.contextValue, context);
+    }
+
+    const uris = await askFolder("Publish");
+    const sourceApimName = node.root.serviceName;
+    const resourceGroup = node.root.resourceGroupName;
+    const subscriptionId = node.root.subscriptionId;
+
+    const noticeContent = localize("Publish", `Publishing '${uris}' to '${sourceApimName}' service.`);
+
+    window.withProgress(
+        {
+            location: ProgressLocation.Notification,
+            title: localize("Extracting", noticeContent),
+            cancellable: false
+        },
+        async () => {
+            await azUtils.checkAzInstalled();
+            await runPublisher(uris[0].fsPath, sourceApimName, resourceGroup, subscriptionId);
+        }
+    ).then(
+        () => {
+            window.showInformationMessage(localize("Published", `Publish completed!`));
+        });
+}
+
+async function runPublisher(filePath: string, apimName: string, resourceGroupName: string, subscriptionId: string): Promise<void> {
+    ext.outputChannel.show();
+
+    // Check our APIOps tooling has been downloaded
+    const downloader = new FileDownloader();
```

Comment:
Looks like we can reuse APIOps downloading, az login and set subscription part between import and export

Reply:
Great idea - created a setSubscription method which `extract` (old DevOps Toolkit), `import` and `export` can re-use. This will assume you are logged in and if not carry out an `az login`